### PR TITLE
Extract feed and item images from more places

### LINKED ISSUE #133: Image always nil
### Expected behavior
Item.Image returns string

### Actual behavior
Item.Image is nil

### Steps to reproduce the behavior
I tried ~2000 articles from 279 different RSS feeds (all of them were XML) and all of the 2000 goffed.Item had nil for image. I looked at the XML feeds, there were image links in them. It must be a parsing errror.

### PR DESCRIPTION
**Pull request by infogulch**
Additional locations where images are attempted to be extracted:

* media:content extension https://www.rssboard.org/media-rss#media-content
* The first `` in content or description

Fixes #133

### CODE REVIEW

File: testdata/translator/rss/feed_image_-_rss_description.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<!--
+Description: channel image
+-->
+<rss version="0.91">
+  <channel>
+    <description><img src="http://example.com/description.png"></description>
```
Comment (infogulch) [inline testdata/translator/rss/feed_image_-_rss_description.xml:6]: Description is parsed as `"</description>"` which doesn't make any sense to me. Is this a bug or expected?

**Comment by infogulch** [comment]
I think this is mostly correct, though I could try to add a few more tests.

File: testdata/translator/rss/feed_image_-_rss_description.xml (hunk 1 above)
Comment (infogulch) [reply testdata/translator/rss/feed_image_-_rss_description.xml:6]: Here's the test output:

```
--- FAIL: TestDefaultRSSTranslator_Translate (0.10s)
    translator_test.go:45: 
                Error Trace:    /home/joe/gofeed/translator_test.go:45
                Error:          Not equal: 
                                expected: &gofeed.Feed{Title:"", Description:"<img src=\"http://example.com/description.png\">", Link:"", FeedLink:"", Links:[]string(nil), Updated:"", UpdatedParsed:<nil>, Published:"", PublishedParsed:<nil>, Author:(*gofeed.Person)(nil), Authors:[]*gofeed.Person(nil), Language:"", Image:(*gofeed.Image)(0xc0004dc440), Copyright:"", Generator:"", Categories:[]string(nil), DublinCoreExt:(*ext.DublinCoreExtension)(nil), ITunesExt:(*ext.ITunesFeedExtension)(nil), Extensions:ext.Extensions(nil), Custom:map[string]string(nil), Items:[]*gofeed.Item{}, FeedType:"rss", FeedVersion:"0.91"}
                                actual  : &gofeed.Feed{Title:"", Description:"<img src=\"http://example.com/description.png\"></description>", Link:"", FeedLink:"", Links:[]string(nil), Updated:"", UpdatedParsed:<nil>, Published:"", PublishedParsed:<nil>, Author:(*gofeed.Person)(nil), Authors:[]*gofeed.Person(nil), Language:"", Image:(*gofeed.Image)(0xc0004dc420), Copyright:"", Generator:"", Categories:[]string(nil), DublinCoreExt:(*ext.DublinCoreExtension)(nil), ITunesExt:(*ext.ITunesFeedExtension)(nil), Extensions:ext.Extensions(nil), Custom:map[string]string(nil), Items:[]*gofeed.Item{}, FeedType:"rss", FeedVersion:"0.91"}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,3 +2,3 @@
                                  Title: (string) "",
                                - Description: (string) (len=46) "<img src=\"http://example.com/description.png\">",
                                + Description: (string) (len=60) "<img src=\"http://example.com/description.png\"></description>",
                                  Link: (string) "",
                Test:           TestDefaultRSSTranslator_Translate
                Messages:       Feed file feed_image_-_rss_channel_description.xml did not match expected output feed_image_-_rss_channel_description.json
    translator_test.go:45: 
                Error Trace:    /home/joe/gofeed/translator_test.go:45
                Error:          Not equal: 
                                expected: &gofeed.Feed{Title:"", Description:"", Link:"", FeedLink:"", Links:[]string(nil), Updated:"", UpdatedParsed:<nil>, Published:"", PublishedParsed:<nil>, Author:(*gofeed.Person)(nil), Authors:[]*go
```

There appears to be an issue with the parser including the closing `</description>` tag as part of the description text.

Comment (mmcdole) [reply testdata/translator/rss/feed_image_-_rss_description.xml]: This seems like a bug.  I'm guessing the pull parser is thrown off by the "self closing" HTML style img tag.  I bet it doesn't mess up if it is `` ?

I haven't looked yet -- but I suspect we are using some convenience method like "ParseText()` function which is supposed to consume the contents of the description tag.  That might be where we have to look.

**Comment by mmcdole** [comment]
@infogulch I think the fallback image sources in the translator function you added look clean and make sense to me, including the HTML parsing code.  I had no clue that many images stash their images in there, lol.

File: testdata/translator/rss/feed_image_-_rss_description.xml (hunk 1 above)
Comment (infogulch) [reply testdata/translator/rss/feed_image_-_rss_description.xml]: You're right, closing the `` tag fixed the parsing behavior.

Comment (infogulch) [reply testdata/translator/rss/feed_image_-_rss_description.xml]: Here? https://github.com/mmcdole/gofeed/blob/master/rss/parser.go#L155-L160

Comment (infogulch) [reply testdata/translator/rss/feed_image_-_rss_description.xml]: So maybe a raw self closing img tag is not a very good test. Looking at the fivethirtyeight.com feed, the description sometimes has html syntax, but rss is an xml format so the description is formatted in a `<![CDATA[...]]>` text block. I think tests that have html content should be CDATA-encoded.

**Comment by mmcdole** [comment]
@infogulch update looks good to me.  

I might create a separate issue to think about what to do with naked HTML markup within tags.

**Comment by mmcdole** [comment]
Thank you for your contribution @infogulch !

Now I just need to tackle #210, and hopefully turn back on gating of PRs for tests passing.

**Comment by spacecowboy** [comment]
I'd like to comment that fetching the first `` inside body isn't such a great idea.

Take for example the feed from slashdot: https://rss.slashdot.org/Slashdot/slashdotMain

The first image in the body will be https://a.fsdn.com/sd/twitter_icon_large.png which is 56x20 pixels. This is directly unsuitable as a thumbnail for an article.

Perhaps it would be better to place the first body image as an `extension`? Then clients can choose if they want to consider it or not?

**Pull request merged**